utils/callData/registrationCalls.ts

```diff
@@ -197,6 +198,14 @@ function multiCallFreeRenewals(encodedDomains: string[]): Call[] {
   return encodedDomains.map((encodedDomain) => freeRenewal(encodedDomain));
 }
 
+function resetAddrToDomain(): Call {
```

Comment:
In which case do you use this in registration flow ?

Reply:
In the case of a user buying a domain, he wants to set it as main domain but already has a domain set as main domain and `addr_has_rev` returns true. We need to reset address to domain or we'll end up in the same case as the bug we got recently.